src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/history/FlowHistory.java

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2023 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowhs.service.history;
+
+
+import static org.openkilda.wfm.share.history.model.DumpType.STATE_AFTER;
+import static org.openkilda.wfm.share.history.model.DumpType.STATE_BEFORE;
+
+import org.openkilda.model.Flow;
+import org.openkilda.model.FlowPath;
+import org.openkilda.wfm.share.history.model.FlowDumpData;
+import org.openkilda.wfm.share.mappers.HistoryMapper;
+
+import lombok.Getter;
+
+@Getter
+public final class FlowHistory {
+    private String flowId;
+    private String action;
+    private String description;
+    private final String taskId;
+    private FlowDumpData flowDumpData;
+
+    private FlowHistory(String taskId) {
+        this.taskId = taskId;
+    }
+
+    public static FlowHistory withTaskId(String taskId) {
```

Comment:
IIRC, java conventions suggest using `of` word for static constructors. So I'd expect the one is named `ofTaskId()`.
Also, this would distinguish constructor from the rest pseudo-builder methods (`withX`).
Why won't you use @Builder annotation, by the way?

Reply:
`with` makes the usage fluent, the whole thing looks like a sentence, like: save flow history with action and with dump. This one is made in a form of a static constructor so that the client can't skip adding the task ID and I actually made it the same as other method names on purpose. (in IDE it won't confuse, there is only one option after typing FlowHistory)
Builder annotation does the same, but without fluency and add builder() and build() methods. of() is also possible, probably it's just a personal preference. People make names different sometimes such as `emptyList` in java collecitons

Reply:
I made it of() kind of make sense as well, `save(FlowHistory.of(correlationId).withAction(...`
I left out `taskId` because there is correlation ID passed in in all cases. I don't know why it was called taskId in history, in other places it is called correlation ID